src/dtos/upload-dto.ts

```diff
@@ -0,0 +1,15 @@
+/* eslint-disable import/no-cycle */
+import { Error } from '../models/error';
+
+import { DatasetDTO } from './dataset-dto';
+
+export interface UploadDTO {
+    success: true;
+    dataset: DatasetDTO;
+}
+
+export interface UploadErrDTO {
+    success: boolean;
+    dataset: DatasetDTO;
+    errors: Error[];
```

Comment:
There's a whole bunch of errors which could fall out of Datalake, there's also the obvious there's no file attached, the file isn't actually a CSV file, etc.  We're still working around happy path stuff so the number of possible errors is quite small but I expect this could grow over time.  So having a DTO to hold and return those errors makes good sense.